connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -359,7 +359,7 @@ private void convertMessages(ConsumerRecords<byte[], byte[]> msgs) {
                     new SinkRecord(msg.topic(), msg.partition(),
                             keyAndSchema.schema(), keyAndSchema.value(),
                             valueAndSchema.schema(), valueAndSchema.value(),
-                            msg.offset())
+                            msg.offset(), msg.timestamp(), msg.timestampType())
```

Comment:
Yes, it could be the sentinel value if `timestampType() == NO_TIMESTAMP_TYPE`. If the connector makes sure to check `timestampType()` before using the `timestamp()`, it would be fine, but it's probably better to provide a consistent expectation of an unavailable timestamp being `null`, so I can fix this.